lib/SimpleNonlinearSolve/src/halley.jl

```diff
@@ -15,16 +15,18 @@ A low-overhead implementation of Halley's Method.
   - `autodiff`: determines the backend used for the Jacobian. Defaults to  `nothing` (i.e.
     automatic backend selection). Valid choices include jacobian backends from
     `DifferentiationInterface.jl`.
+  - `taylor_mode`: whether to use Taylor mode automatic differentiation to compute the Hessian-vector-vector product. Defaults to `Val(false)`. If `Val(true)`, you must have `TaylorDiff.jl` loaded.
```

Comment:
This is going to be extremely confusing and I don't see why can't there be a `AutoTaylorDiff()` even if it is being used for very restrictive usecases?

For example, what does it mean if someone specifies AutoForwardDiff + tyalor_mode = Val(true). In other parts of SciML, re OrdinaryDiffEq we are aggressively moves away from this kind of multi-arg AD specifications, it would be a bad idea to re-introduce that here.

Reply:
A better way would be to use SecondOrder from DifferentiationInterface to specify using different backends. If outer is AutoTaylorDiff then you use the specialized version

Reply:
I really think we should just make TaylorDiff use ADTypes. If we don't use DifferentiationInterface.jl here then that's a detail, but I don't think that should effect the user interface.

Reply:
The reason is that currently TaylorDiff haven't got chunk size which makes it less efficient to compute the Jacobian. So, even if the user choose to use TaylorDiff for the Halley correction, they should still use ForwardDiff or something else to do the Jacobian.

If there is only one argument to be AutoTaylorDiff(), then I'm not sure what to do with Jacobian.

Reply:
Just hardcode it then.

Reply:
By saying "hardcode", did you mean when argument is "AutoTaylorDiff()", the behavior is to use TaylorDiff for Halley correction and ForwardDiff for Jacobian?

Reply:
yes

Reply:
The taylor_mode arg is removed.